tests/tests/oob_indexing.rs

```diff
@@ -230,3 +230,227 @@ impl TestResources {
         }
     }
 }
+
+/// Tests behavior of OOB accesses for dynamic buffers.
+///
+/// This test is specific to D3D12 since Vulkan and Metal behave differently and
+/// the WGSL spec allows for multiple behaviors when it comes to OOB accesses.
+#[gpu_test]
+static RESTRICT_DYNAMIC_BUFFERS: GpuTestConfiguration = GpuTestConfiguration::new()
+    .parameters(
+        TestParameters::default()
+            .downlevel_flags(wgpu::DownlevelFlags::COMPUTE_SHADERS)
+            .limits(wgpu::Limits::downlevel_defaults())
+            .skip(FailureCase::backend(Backends::all() - Backends::DX12)),
+    )
+    .run_async(|ctx| async move {
```

Comment:
Nit: can we move the body of the test into its own function. This helps RA a lot, as it gets kinda confused by the macro. I've been idly musing about being able to get rid of the macro to not much success.